drivers/flash/Kconfig.stm32

```diff
@@ -7,7 +7,7 @@
 config SOC_FLASH_STM32
 	bool "STM32 flash driver"
 	depends on SOC_FAMILY_STM32
-	depends on (SOC_SERIES_STM32F0X || SOC_SERIES_STM32F1X || SOC_SERIES_STM32F2X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L0X  || SOC_SERIES_STM32L1X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32L5X || SOC_SERIES_STM32WBX || SOC_SERIES_STM32WLX || SOC_SERIES_STM32G0X || SOC_SERIES_STM32G4X || SOC_SERIES_STM32H7X)
+	depends on (SOC_SERIES_STM32F0X || SOC_SERIES_STM32F1X || SOC_SERIES_STM32F2X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L0X  || SOC_SERIES_STM32L1X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32L5X || SOC_SERIES_STM32U5X || SOC_SERIES_STM32WBX || SOC_SERIES_STM32WLX || SOC_SERIES_STM32G0X || SOC_SERIES_STM32G4X || SOC_SERIES_STM32H7X)
```

Comment:
Not blocking: would be nice to replace this line by a :
```
DT_COMPAT_ST_STM32_FLASH_CONTROLLER := st,stm32-flash-controller
[...]
	depends on $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FLASH_CONTROLLER))
```

Reply:
Note: This might need some tests. I'm not sure how this will behave when there is no flash controller available for the target.

Reply:
ok

Reply:
with a special case for H7 : `DT_COMPAT_ST_STM32H7_FLASH_CONTROLLER := st,stm32h7-flash-controller`